packages/xo-server/src/xo-mixins/backups-ng/index.js

```diff
@@ -1175,7 +1175,7 @@ export default class BackupNg {
       }
     } else {
       const snapshots = vm.$snapshots
-        .filter(_ => _.other_config['xo:backup:job'] === jobId)
+        .filter(_ => _.other_config['xo:backup:schedule'] === scheduleId)
```

Comment:
I'm sorry for not having this sooner but this change is completely wrong.

`snapshots` must contains all snapshots related to this job because it is used to find the more recent base in case of a delta.

Also, current code appears to already be correct because it already handle snapshots by schedule for removal, please see below.

Reply:
I see, the idea between fetching all job's snapshots is to handle the case when it has two schedules dedicated to the delta backup.

 It's an optimization to only export the diff between the last exported state and the current state.

Unfortunately, this implementation has a limitation. As evoked, in the forum discussion linked in the PR description. A user want a schedule dedicated to snapshots and a schedule dedicated to delta backup. With the current implementation, is not possible to do it. Rolling snapshot will delete the snapshot created by the delta backup.